analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantArgument.cs

```diff
@@ -52,10 +51,11 @@ protected override void Initialize(SonarAnalysisContext context)
                         return;
                     }
 
-                    var methodCall = (InvocationExpressionSyntax) c.Node;
-                    var methodParameterLookup = new CSharpMethodParameterLookup(methodCall, c.SemanticModel);
-                    var argumentMappings = methodParameterLookup.GetAllArgumentParameterMappings()
-                        .ToList();
+                    var argumentList = c.Node is InvocationExpressionSyntax invocationExpression
+                        ? invocationExpression.ArgumentList
+                        : ((ImplicitObjectCreationExpressionSyntaxWrapper)c.Node).ArgumentList;
+                    var methodParameterLookup = new CSharpMethodParameterLookup(argumentList, c.SemanticModel);
+                    var argumentMappings = methodParameterLookup.GetAllArgumentParameterMappings().ToList();
 
                     var methodSymbol = methodParameterLookup.MethodSymbol;
                     if (methodSymbol == null)
```

Comment:
While we're touching this rule, let's remove the return and make it shorter
```
if(methodParameterLookup.MethodSymbol is {} methodSymbol)
{
    // Move var argumentMappings = .. here as a performance optimization - it's not needed before the condition
}

Reply:
Or even better - rename the foreach variable to `x` to make it short. And then we don't really need `argumentMappings` variable nor the `ToList()`

Reply:
I'd be also fine with removing `argument` and `parameter` as it's just a rename (waste of lines)